src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/HmrcPageHeading.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hmrcfrontend.views.config
+
+import play.api.i18n.Messages
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.govukfrontend.views.Utils.toClasses
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.HtmlContent
+import uk.gov.hmrc.govukfrontend.views.viewmodels.fieldset.Legend
+import uk.gov.hmrc.govukfrontend.views.viewmodels.label.Label
+
+trait HmrcPageHeadingConfig {
+  protected val defaultVisuallyHiddenText = "This section is"
+
+  protected def captionFor(section: String, visuallyHiddenText: Option[String])(implicit messages: Messages): String =
+    s"""<span class="govuk-caption-xl hmrc-caption-xl">${visuallyHiddenText match {
+      case Some(hiddenText) =>
+        s"""<span class="govuk-visually-hidden">${HtmlFormat.escape(hiddenText match {
+          case `defaultVisuallyHiddenText` => messages("hmrcPageHeading.visuallyHiddenSectionPrefix")
```

Comment:
I had to look up what back-ticks do in pattern matching :-) Does this logic mean if the text 'This section is' is explicitly provided in English, it will be translated to Welsh automatically? I wonder if this is what we want, feels a bit surprising. Couldn't we just make None the default and match against that?

Reply:
Yeah it matches against the content of the variable, if we match against none then we can't give the ability to omit the visually hidden text, but maybe we don't need that - I'm not sure if there's a better way to structure this so we have a translated default and teams can override it themselves for specific usages

Reply:
Just realised I think could remove the default values and provide an alternate constructor with no visuallyHiddenText parameter that passed in the message, I'll try that out

Reply:
Worth considering if we even want this to be overridable if it's part of the pattern?

Reply:
I have seen services which have changed this text because it didn't make sense in the context of the heading.

Reply:
@matthewmascord I changed the api from `visuallyHiddenText: Option[String]` to `visuallyHiddenText: Option[StringOrMessage]` and then I interpret this inline using the implicit messages to the right value. So the default can be `Some(StringOrMessage.StringFromMessages("hmrcPageHeading.visuallyHiddenSectionPrefix")` and if someone wants to override it they can use their own string from messages or they can use `Some(StringOrMessage.InlineString("Section"))` instead. I put the case class in viewmodels but let me know if you think it should be somewhere better. It feels to me like it's a good idea to have a translation by default rather than a fixed english string even if it is overridable because it would be very easy to miss that a translation was needed for this because it's visually hidden and you wouldn't know it's there unless you wanted something different to the default.

Reply:
then if we wanted we could have an implicit conversion that went from Option[String] -> Option[StringOrMessage.InlineString] to save some extra verboseness